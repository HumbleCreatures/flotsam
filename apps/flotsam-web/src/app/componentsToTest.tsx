export const PropLogicComponentToTest = ({message}:{ message?:string }) => {
  if(message) {
    return <div>{message}</div>
  }

  return <><div>There is no message again</div><div>Middle message</div><div>Third message</div></>
}

export const EventComponentToTest = ({message, doStuff}:{message:string, doStuff:(input:string)=>void}) => {
  return <button onClick={() => {doStuff(message)}}></button>
}


