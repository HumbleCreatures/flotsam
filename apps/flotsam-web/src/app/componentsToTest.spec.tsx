import { fireEvent, render } from '@testing-library/react';
import { EventComponentToTest, PropLogicComponentToTest } from './componentsToTest'


describe('Test a component that has different logic depending on a prop', () => {
  it('should display message if set', () => {
    const {getByText} = render(<PropLogicComponentToTest message='new message' />);
    const element = getByText('new message');
    expect(element).toBeTruthy();
  });

  it('should display default message if message is not set', () => {
    const {getByText} = render(<PropLogicComponentToTest />);
    const element = getByText('There is no message again');
    const thirdElement = getByText('Third message');
    expect(thirdElement).toBeTruthy();
    expect(element).toBeTruthy();
  });

  it('should display default message if message is not set with snapshot', () => {
    const {baseElement} = render(<PropLogicComponentToTest />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe('Test a component with a click event', () => {
  it('should have an a function that has been called with a value', () => {
    const funcToGetCalled = jest.fn();
    const {getByRole} = render(<EventComponentToTest message='a message' doStuff={funcToGetCalled} />);
    const button = getByRole('button');
    fireEvent.click(button);
    expect(funcToGetCalled).toBeCalled();
    expect(funcToGetCalled).toBeCalledWith('a message');
  });
});


