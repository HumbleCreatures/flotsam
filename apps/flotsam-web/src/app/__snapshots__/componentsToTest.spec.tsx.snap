// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test a component that has different logic depending on a prop should display default message if message is not set with snapshot 1`] = `
<body>
  <div>
    <div>
      There is no message again
    </div>
    <div>
      Another message
    </div>
    <div>
      Third message
    </div>
  </div>
</body>
`;
